### ISSUE TITLE
predicted probabilities not summed to 1

### ISSUE THREAD

**Issue by jpainam** [shaohua0116/SSGAN-Tensorflow#19]
Hi!
Pls, can you see this @shaohua0116 @gitlim @Shaofanl @carpedm20 
I tried this
` predictions = np.argmax(self._predictions[-1], axis=0)`
or this
`predictions = np.argmax(prediction_pred, axis=0)`
and i get
`[39 38 59 45 49  1  8 30 25 44 53]`
These probabilities do not summed to 1. Am I doing something wrong?
prediction_pred comes from the softmax Discriminator layer

**Comment by MillPopulus** [comment]
Because this code uses sigmoid instead of softmax，and I suppose sigmoid may be wrong here.

https://github.com/gitlimlab/SSGAN-Tensorflow/blob/1323a53f02503caa28b67e1ee52a4c30252a51dd/discriminator.py#L37

**Comment by shaohua0116** [comment]
Note that we actually do not use the sigmoid output to compute the loss for the discriminator. Instead, we use the logits with `tf.nn.softmax_cross_entropy_with_logits`. To get the predicted probabilities, you can apply `softmax` on the last dimension of the output logits.
